cmd/cluster/dynatrace.go

```diff
@@ -42,16 +42,26 @@ func fetchDetails(clusterID string) error {
 	}
 	defer connection.Close()
 
-	cluster, err := ocmutils.GetCluster(connection, clusterID)
+	cluster, err := ocmutils.GetCluster(connection, clusterKey)
 	if err != nil {
 		return err
 	}
 
-	if !isManagementCluster(connection, cluster) {
-		return fmt.Errorf("cluster is not a management cluster")
+	clusterID = cluster.ID()
+	// Get MC Cluster if it is HCP
+	if cluster.Hypershift().Enabled() {
+		ManagementCluster, err := ocmutils.GetManagementCluster(cluster.ID())
+		if err != nil {
+			return fmt.Errorf("error retreiving Management Cluster for given HCP")
+		}
+		clusterID = ManagementCluster.ID()
+	}
+
+	if !isManagementCluster(connection, clusterID) && !cluster.Hypershift().Enabled() {
+		return fmt.Errorf("cluster is not a HCP/Management Cluster")
```

Comment:
This logic is duplicate, would it make sense to move it into `GetDynatraceURLFromCluster()`?

Reply:
Refactored the logic a bit to reduce duplicate